openapi/spec-generator/specbuilder.go

```diff
@@ -0,0 +1,490 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/getkin/kin-openapi/openapi3gen"
+	"github.com/ghodss/yaml"
+)
+
+// Spec wraps a kin-openapi document object model with extra
+// metadata so that the template can be entirely data driven
+type Spec struct {
+	ValidationContext context.Context // Required parameter for validation functions
+	Model             openapi3.T      // Document object model we are building
+}
```

Comment:
Is this struct needed? Should we just return `openapi3.T`? ValidationContext isn't something callers are going to use and is always `context.Background()`.

Reply:
Refactored out the ValidationContext, but kept the struct for the ToBytes and ToYAML convenience methods, per our discussion.